iceoryx_binding_c/test/moduletests/test_notification_info.cpp

```diff
@@ -250,4 +256,162 @@ TEST_F(iox_notification_info_test, callbackCanBeCalledMultipleTimes)
+    constexpr uint64_t ARBITRARY_EVENT_ID = 124U;
+    ASSERT_FALSE(m_waitSet.attachEvent(m_userTrigger, ARBITRARY_EVENT_ID).has_error());
+    m_userTrigger.trigger();
+
+    auto notificationInfoVector = m_waitSet.wait();
+
+    EXPECT_TRUE(iox_notification_info_does_originate_from_user_trigger(notificationInfoVector[0U], &m_userTrigger));
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>(
+        [&] { iox_notification_info_does_originate_from_user_trigger(nullptr, &m_userTrigger); },
+        iox::HoofsError::EXPECTS_ENSURES_FAILED);
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>(
+        [&] { iox_notification_info_does_originate_from_user_trigger(notificationInfoVector[0U], nullptr); },
+        iox::HoofsError::EXPECTS_ENSURES_FAILED);
+}
+
+TEST_F(iox_notification_info_test, doesOriginateFromClientWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "a8897441-3af6-4d26-9d83-a52b9bd3e148");
+    GTEST_SKIP() << "@todo: iox-1106 to fix later";
+    iox_client_storage_t sutStorage;
+    iox_client_t sut = iox_client_init(&sutStorage, "TheHoff", "IsALL", "YouNeed", nullptr);
+    constexpr uint64_t ARBITRARY_EVENT_ID = 124U;
+    ASSERT_FALSE(m_waitSet.attachEvent(m_userTrigger, ARBITRARY_EVENT_ID).has_error());
+    m_userTrigger.trigger();
+
+    auto notificationInfoVector = m_waitSet.wait();
+    EXPECT_TRUE(iox_notification_info_does_originate_from_client(notificationInfoVector[0U], sut));
+}
+
+TEST_F(iox_notification_info_test, doesOriginateFromClientWithNullptrFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "953b661d-d1f6-47bf-a113-96559f1492aa");
+    GTEST_SKIP() << "@todo: iox-1106 to fix later";
+    // prepareClientInit();
+    iox_client_options_t options;
+    iox_client_options_init(&options);
+    options.responseQueueCapacity = 456;
+    strncpy(options.nodeName, "hypnotoad is all you need", IOX_CONFIG_NODE_NAME_SIZE);
+    options.connectOnCreate = false;
+    options.responseQueueFullPolicy = QueueFullPolicy_BLOCK_PRODUCER;
+    options.serverTooSlowPolicy = ConsumerTooSlowPolicy_WAIT_FOR_CONSUMER;
+    options.initCheck = 47113130815;
+
+    iox_client_storage_t sutStorage;
+    iox_client_t sut = iox_client_init(&sutStorage, "TheHoff", "IsALL", "YouNeed", &options);
+    constexpr uint64_t ARBITRARY_EVENT_ID = 124U;
+    ASSERT_FALSE(m_waitSet.attachEvent(m_userTrigger, ARBITRARY_EVENT_ID).has_error());
+    m_userTrigger.trigger();
+
+    auto notificationInfoVector = m_waitSet.wait();
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>([&] { iox_notification_info_does_originate_from_client(nullptr, sut); },
+                                              iox::HoofsError::EXPECTS_ENSURES_FAILED);
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>(
+        [&] { iox_notification_info_does_originate_from_client(notificationInfoVector[0U], nullptr); },
+        iox::HoofsError::EXPECTS_ENSURES_FAILED);
+    iox_client_deinit(sut);
```

Comment:
This can potentially be simplified. You do not need a valid pointer but only an non-nullptr for this test since it is expected to fail early anyway. I'm currently just not sure if it is a good idea or whether is is easy to get such a pointer for this structs like it is done for the subscriber